continuous_integration/scripts/test_report.py

```diff
@@ -140,17 +141,48 @@ def get_jobs(run, session):
     # Somehow the job ID is not part of the workflow schema and we have no other way to later join
     # this to the JXML results.
 
-    name_components = df_jobs.name.str.split("-", expand=True).dropna()
-    if len(name_components.columns) != 5:
-        raise ValueError(f"Job names must have 5 components:\n{name_components!r}")
-    name_components.columns = ["OS", "OS version", "environment", "label", "partition"]
+    if repo.endswith("/dask"):
+        # example name: "test (windows-latest, 3.9)" or "test (ubuntu-latest, 3.12, dask-expr)"
+        df_jobs = df_jobs[~df_jobs.name.str.contains("Event File")]
+        name_components = (
+            df_jobs.name.str.replace("(", ",")
+            .str.replace(")", ",")
+            .str.split(",", expand=True)
+            .apply(
+                lambda row: ",".join(
+                    (row[1].strip(), row[2].strip(), row[3].strip() or "-")
+                ),
+                axis=1,
+            )
+            .str.split(",", expand=True)
+        )
 
-    # See `Set $TEST_ID` step in `tests.yaml`
-    name_components["partition"] = name_components.partition.str.replace(" ", "")
+        # See `Set $TEST_ID` step in `tests.yaml`
+        name_components.columns = ["OS version", "environment", "option"]
+        df_jobs["suite_name"] = name_components.iloc[:, 0].str.cat(
+            name_components.iloc[:, 1:], sep="-"
+        )
+        df_jobs["suite_name"] = ("test-results-" + df_jobs.suite_name).str.replace(
+            "--", "-"
+        )
```

Comment:
```suggestion
        df_jobs["suite_name"] = df_jobs.suite_name.str.replace("--", "-")
```
Adjust to https://github.com/dask/dask/pull/10781